django/test/testcases.py

```diff
@@ -1030,8 +1030,18 @@ def _fixture_teardown(self):
                          allow_cascade=self.available_apps is not None,
                          inhibit_post_migrate=inhibit_post_migrate)
 
-    def assertQuerysetEqual(self, qs, values, transform=repr, ordered=True, msg=None):
+    def assertQuerysetEqual(self, qs, values, transform=None, ordered=True, msg=None):
+        values = list(values)
+
+        if values:
+            if transform is None:
+                if isinstance(values[0], str):
+                    transform = repr
```

Comment:
I think we should `warnings.warn('...',  category=RemovedInDjango40Warning)` here.